ui/helpers/utils/token-util.js

```diff
@@ -261,17 +261,15 @@ export async function getAssetDetails(
     // if we can't determine any token standard or details return the data we can extract purely from the parsed transaction data
     return { toAddress, tokenId };
   }
-
+  const tokenValue = getTokenValueParam(tokenData);
+  const tokenDecimals = tokenDetails?.decimals;
   const tokenAmount =
     tokenData &&
-    tokenDetails?.decimals &&
-    calcTokenAmount(
-      getTokenValueParam(tokenData),
-      tokenDetails?.decimals,
-    ).toString(10);
-
-  const decimals =
-    tokenDetails?.decimals && Number(tokenDetails.decimals?.toString(10));
+    tokenValue &&
+    tokenDecimals &&
```

Comment:
How does this handle tokens with `decimals` set to `0`?

Reply:
As far as i can see, 0 decimals is handled by the usage of tokenDecimals.
- inside `calcTokenAmount`, decimal 0 is accepted 
```
  const multiplier = Math.pow(10, Number(decimals || 0));
```
- inside the decimals variable, it doesn't seem like 0 would impact the final result 
```
  const decimals = tokenDecimals && Number(tokenDecimals?.toString(10));
```
do you have any other concerns i didn't catch?

Reply:
Ah I see, this is a string right? I was thinking that if this was a number, 0 would be falsy and `calcTokenAmount` would never be run. But if it's a string, this should work fine.